has_single_bit requires #include <bit>

https://github.com/rui314/mold/blob/0d542fb3bc9553fc3e48558a8c22b6b08d3d1d85/mold.h#L163

Introduced by 6c9bb0d7ff5e863775e50f5a80fe64cefff34a23

Building on musl with LLVM/Clang/lld/libc++.

Adding `#include <bit>` to `mold.h` resolves the below build time error:

```
mkdir -p out/elf
touch out/elf/.keep
mkdir -p out/macho
touch out/macho/.keep
c++ -std=c++20 -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables -DMOLD_VERSION=\"1.0.2\" -DLIBDIR="\"/usr/local/lib\"" -Ithird-party/mimalloc/include -Ithird-party/tbb/include -Ithird-party/xxhash  -MT out/compress.o -MMD -MP -MF out/compress.d -O2 -c -o out/compress.o compress.cc
In file included from compress.cc:15:
./mold.h:163:15: error: no member named 'has_single_bit' in namespace 'std'
  assert(std::has_single_bit(align));
         ~~~~~^
/usr/include/assert.h:8:28: note: expanded from macro 'assert'
#define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
                           ^
In file included from compress.cc:15:
./mold.h:168:15: error: no member named 'has_single_bit' in namespace 'std'
  assert(std::has_single_bit(align));
         ~~~~~^
/usr/include/assert.h:8:28: note: expanded from macro 'assert'
#define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
                           ^
In file included from compress.cc:15:
./mold.h:299:49: error: no member named 'bit_ceil' in namespace 'std'
    nbuckets = std::max<i64>(MIN_NBUCKETS, std::bit_ceil<u64>(nbuckets));
                                           ~~~~~^
./mold.h:299:58: error: unexpected type name 'u64': expected expression
    nbuckets = std::max<i64>(MIN_NBUCKETS, std::bit_ceil<u64>(nbuckets));
                                                         ^
./mold.h:311:17: error: no member named 'has_single_bit' in namespace 'std'
    assert(std::has_single_bit<u64>(nbuckets));
           ~~~~~^
/usr/include/assert.h:8:28: note: expanded from macro 'assert'
#define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
                           ^
In file included from compress.cc:15:
./mold.h:311:32: error: unexpected type name 'u64': expected expression
    assert(std::has_single_bit<u64>(nbuckets));
                               ^
./mold.h:429:57: error: no member named 'countl_zero' in namespace 'std'
    update_maximum(buckets[hash & (NBUCKETS - 1)], std::countl_zero(hash) + 1);
                                                   ~~~~~^
7 errors generated.
gmake: *** [Makefile:140: out/compress.o] Error 1
```

--- a/mold.h
+++ b/mold.h
@@ -3,6 +3,7 @@
 #include "byteorder.h"
 
 #include <atomic>
+#include <bit>
 #include <cassert>
 #include <cstdio>
 #include <cstring>

